pandas/tests/frame/test_timeseries.py

```diff
@@ -374,6 +374,31 @@ def test_truncate_copy(self):
         truncated.values[:] = 5.
         assert not (self.tsframe.values[5:11] == 5).any()
 
+    def test_truncate_nonsortedindex(self):
```

Comment:
can you add a test for series as well (you can also put this test generically in test_generic if its easier)